receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -129,17 +130,18 @@ func (tsm *timeseriesMap) get(metric pmetric.Metric, kv pcommon.Map) (*timeserie
 	return tsi, ok
 }
 
-// Create a unique timeseries signature consisting of the metric name and label values.
+// Create a unique string signature for attributes.
 func getAttributesSignature(kv pcommon.Map) string {
-	labelValues := make([]string, 0, kv.Len())
-	kv.Sort().Range(func(_ string, attrValue pcommon.Value) bool {
+	attrs := make([]string, 0, kv.Len())
+	kv.Range(func(key string, attrValue pcommon.Value) bool {
 		value := attrValue.Str()
 		if value != "" {
-			labelValues = append(labelValues, value)
+			attrs = append(attrs, key+"="+value)
 		}
 		return true
 	})
-	return strings.Join(labelValues, ",")
+	sort.Strings(attrs)
+	return strings.Join(attrs, ",")
 }
```

Comment:
Key was not there, it is confusing this PR.

Reply:
Also where is sort called?

Reply:
> Key was not there, it is confusing this PR.

I described why this change is made in the PR description. It doesn't change the functionality, just makes the signature longer. 

> Also where is sort called?

`sort.Strings(attrs)` is called on the key-value pairs giving the same order as previously

Reply:
From the description:

> We could keep the existing signature representation, but that would require creating extra entities like a cloned pcommon.Map or a custom key-value slice. I believe it's ok to include attribute names in the signature instead to avoid those complications.

Please let me know if you think we should do differently

Reply:
I now understand better the function, but you are doing something different. Can you provide a solution that does not require changing the behavior of the func which is that the arg keys are not present.

Reply:
Updated the func to keep the existing signature representation. It's a bit more complicated now. PTAL